### ISSUE TITLE
Issues passing `schema` to `write_delta`

### ISSUE THREAD

**Issue by hawkaa** [pola-rs/polars#10798]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
    import pyarrow as pa
    import polars as pl
    
    schema = pa.schema(
        [
            pa.field("time", pa.timestamp("ns", tz="UTC"), nullable=False),
        ]
    )
    df = pl.DataFrame(
        {"time": [datetime(2023, 8, 30, tzinfo=timezone.utc)]},
    )
    df.write_delta("lol/", delta_write_options={"schema": schema})
```
Gives the following output:

```
  File "/Users/hakon/Library/Caches/pypoetry/virtualenvs/zora-harvester-8BY1SBbk-py3.11/lib/python3.11/site-packages/polars/dataframe/frame.py", line 3478, in write_delta
    write_deltalake(
TypeError: deltalake.writer.write_deltalake() got multiple values for keyword argument 'schema'
```

### Issue description

According to the `DataFrame.write_delta` docstring, this is the correct way to enforce a schema with nullable values. However, it doesn't work. Looking at the source code, it looks like there's a problem how `schema` is handled.

```python
def write_delta(...):
    ...
        schema = delta_write_options.get("schema")
        if schema is None:
            schema = _convert_pa_schema_to_delta(data.schema)

        data = data.cast(schema)

        write_deltalake(
            table_or_uri=target,
            data=data,
            schema=schema,
            mode=mode,
            overwrite_schema=overwrite_schema,
            storage_options=storage_options,
            **delta_write_options,
        )
 ```
 
 As you can see, `schema` is being pulled out of `delta_write_options` and converted. However, it is never deleted from `delta_write_options`, hence it is included twice in the call to `write_deltalake` (one as delta schema, one as pyarrow schema)

### Expected behavior

The delta table should be written and enforce nullable columns.

### Installed versions

<details>

```
--------Version info---------
Polars:              0.18.15
Index type:          UInt32
Platform:            macOS-13.4.1-x86_64-i386-64bit
Python:              3.11.4 (v3.11.4:d2340ef257, Jun  6 2023, 19:15:51) [Clang 13.0.0 (clang-1300.0.29.30)]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           0.10.1
fsspec:              <not installed>
matplotlib:          <not installed>
numpy:               1.25.2
pandas:              <not installed>
pyarrow:             13.0.0
pydantic:            <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

**Comment by ion-elgreco** [comment]
It's already fixed: https://github.com/pola-rs/polars/pull/10541/files

**Comment by stinodego** [comment]
We just released `0.19.0`, which includes the fix.
